.gitlab-ci.yml

```diff
@@ -0,0 +1,84 @@
+---
+stages:
+  - 'validation'
+
+variables:
+  PUPPET_VERSION:    'UNDEFINED' # <- Matrixed jobs MUST override this (or fail)
```

Comment:
> `Matrixed jobs MUST override this (or fail)`

This is true when `PUPPET_VERSION` is `UNDEFINED`

Reply:
For the time being, I haven't put in any additional matrix data. I was going to suggest simply removing it but I haven't been able to see a gitlab-ci evaluate this change.

Reply:
I'd remove `PUPPET_VERSION`; the Gemfile doesn't use it.

Before running in gitlab-ci, you can still test what will happen locally by setting the `PUPPET_VERSION` environment variable while running the steps in each job, e.g,:

```sh
export PUPPET_VERSION=UNDEFINED # or `unset PUPPET_VERSION`
bundle update
bundle exec rake fingerprint
```